### PR TITLE
Add icon for broken layer

### LINKED ISSUE #410: Show a "didn't load" broken icon next to a layer if it fails to fetch data
Sometimes layers fail to fetch data from its API. In such cases show a broken icon next to the layer in the layer menu to visually indicate that the data fetch was not successful.

> We could separately decide to show a "didn't load" broken icon like https://fontawesome.com/v4.7.0/icon/exclamation-triangle next to the layer name!

Quoting @jywarren from https://github.com/publiclab/leaflet-environmental-layers/issues/290#issue-532072282

### PR DESCRIPTION
**Pull request by crisner**
Fixes #410  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by crisner** [comment]
image

The icon gets removed when we turn off the layer.

**Comment by crisner** [comment]
I have added this feature for all layers that use ajax to get data. I haven't yet figured out how to implement this in layers that have tile data. There seem to be some errors that need to get fixed.

**Comment by crisner** [comment]
I've added this feature to almost all layers. Unable to get it working in a couple of layers in Open Weather Map(cities and windrose) layer and PLpeople. @nstjean, any idea how I can detect an error in the PLpeople layer. I turned off my network while testing but PLpeople doesn't seem to be throwing any errors in the console.

**Comment by nstjean** [comment]
Hmmm, good question. It looks like you'll have to go into LeafletBlurredLocationDisplay, where it's calling the api, and set up an error on fail:
 https://github.com/publiclab/leaflet-environmental-layers/blob/43a11e3e40ac4967dd0b1ec42cc157745b7123fa/src/PLpeopleLayer.js#L30

**Comment by crisner** [comment]
Thanks! I'll look into it.

**Comment by crisner** [comment]
I have made a workaround for the layers in OWM I was having issues with. I have held off on implementing this feature in PLpeople as I couldn't find a way to do it in the layer's file.

File: cypress/integration/layerErrorIcon.spec.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+describe('Broken icon for layer errors', function() {
```
Comment (crisner) [inline cypress/integration/layerErrorIcon.spec.js]: @sagarpreet-chadha, is there a way you are aware of where I can make a stub send a request error in its response. I want to try to mimic xhr errors but I can't seem to figure out how.

File: cypress/integration/layerErrorIcon.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  it('shows broken icon on a layer error when layer is added to map', function() {
+    cy.openWindow('/example/index.html#lat=41.624&lon=-91.626&zoom=5&layers=Standard')
+    cy.wait(300)
+    cy.window().then((win) => {
+      cy.stub(win.FracTracker_mobile, 'requestData')
+      cy.get('#menu-fracTrackerMobile label').click({ force: true }).then(() => {
+        win.FracTracker_mobile.onError('fracTrackerMobile')
```
Comment (crisner) [inline cypress/integration/layerErrorIcon.spec.js]: I checked the feature by stubbing the layer and manually calling the method instead.

**Comment by jywarren** [comment]
Oh, small issue, this needs to be rebuilt with latest main branch merged in... @crisner are you able to do this? Thank you!!!

**Pull request merged**